source/scripts/fetch-stats.js

```diff
@@ -0,0 +1,333 @@
+// This script uses the Matomo API which requires an access token
+// Once you have your access token you can put it in a `.env` file at the root
+// of the project to enable it during development. For instance:
+//
+// MATOMO_TOKEN=f4336c82cb1e494752d06e610614eab12b65f1d1
+//
+// Matomo API documentation:
+// https://developer.matomo.org/api-reference/reporting-api
+
+require('dotenv').config()
+require('isomorphic-fetch')
+const querystring = require('querystring')
+const { createDataDir, writeInDataDir } = require('./utils.js')
+const R = require('ramda')
+
+const apiURL = params => {
+	const query = querystring.stringify({
+		period: 'month',
+		date: 'last1',
+		method: 'API.get',
+		format: 'JSON',
+		module: 'API',
+		idSite: 39,
+		language: 'fr',
+		apiAction: 'get',
+		token_auth: process.env.MATOMO_TOKEN,
+		...params
+	})
+	return `https://stats.data.gouv.fr/index.php?${query}`
+}
+
+async function main() {
+	createDataDir()
+	const stats = {
+		simulators: await fetchSimulatorsMonth(),
+		monthly_visits: await fetchMonthlyVisits(),
+		daily_visits: await fetchDailyVisits(),
+		status_chosen: await fetchStatusChosen(),
+		feedback: await fetchFeedback()
+	}
+	writeInDataDir('stats.json', stats)
+}
+function firstDayXMonthAgo(dt, X = 0) {
+	let [year, month] = dt.split('-')
+	if (month - X > 0) {
+		month -= X
+	} else {
+		month = 12 + (month - X)
+		year -= 1
+	}
+	const pad = n => (+n < 10 ? `0${n}` : '' + n)
+	return {
+		date: `${year}-${pad(month)}-01`,
+		monthyear: `${pad(month)}-${year}`
+	}
+}
+async function fetchSimulatorsMonth() {
+	const today = new Date().toJSON().slice(0, 10)
+	const months = {
+		currentMonth: {
+			date: firstDayXMonthAgo(today, 0).monthyear,
+			visites: await fetchSimulators(today)
+		},
+		oneMonthAgo: {
+			date: firstDayXMonthAgo(today, 1).monthyear,
+			visites: await fetchSimulators(firstDayXMonthAgo(today, 1).date)
+		},
+		twoMonthAgo: {
+			date: firstDayXMonthAgo(today, 2).monthyear,
+			visites: await fetchSimulators(firstDayXMonthAgo(today, 2).date)
+		}
```

Comment:
```suggestion
	const months = 
		R.pipe(
			R.map(xMonthsAgo => firstDayXMonthAgo(today, xMonthsAgo)),
			R.map(async ({monthyear, date}) => ({
					date: monthyear,
					visites: await fetchSimulators(date)
			}))
		)({
			currentMonth: 0,
			oneMonthAgo: 1,
			twoMonthAgo: 2
		})
```

Tu peux aussi essayer quelque chose comme ça pour éviter de dupliquer le code.